cmd/notary/delegations.go

```diff
@@ -188,6 +193,11 @@ func (d *delegationCommander) delegationAdd(cmd *cobra.Command, args []string) e
 		}
 	}
 
+	// If the user passes --all-paths, give the "" path
+	if allPaths {
```

Comment:
It might be ok to only do "smart logic" around the "all paths" exception. If you add `""` and other stuff, ignore the other stuff (or remove it). If you delete `""` also make sure everything else is empty.

We don't have to do more smart logic than that. Not sure what @endophage things, but I would rather do that VS having another bool on the api.